lib/transport/transport-socket.c

```diff
@@ -124,13 +124,7 @@ _parse_cmsg_to_aux(LogTransportSocket *self, struct msghdr *msg, LogTransportAux
 static gssize
 _extract_from_msghdr_method(LogTransportSocket *self, gint rc, struct msghdr *msg, LogTransportAuxData *aux)
 {
-  if (rc == 0)
-    {
-      /* DGRAM sockets should never return EOF, they just need to be read again */
-      rc = -1;
-      errno = EAGAIN;
-    }
-  else if (rc > 0)
+  if (rc > 0)
```

Comment:
The signature of this function suggests, that it might change the `rc`, which is not the case. I think it would be cleaner if we moved the `rc > 0` to the call site, make this a `void` function, and remove `rc` from its arguments.